networkconfig/api/v1/networkconfig_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// AvailableIPMapping groups an IP address and Name. This is a struct and not a Map for future extensibility
+type AvailableIPMapping struct {
+	Name string `json:"name,omitempty"`
+	IP   string `json:"ip,omitempty"`
+}
+
+// NetworkContainerSpec defines the structure of a Network Container as found in NetworkConfigStatus
+type NetworkContainerSpec struct {
+	PrimaryIP      string               `json:"primaryIP,omitempty"`
```

Comment:
Will CNS consume anythign besides the available IPs? if not should we be yagni and remove remove some of these  until their needed to keep the interface thin?

Reply:
CNS will be consuming all of these fields